libs/FScreen.c

```diff
@@ -409,9 +409,13 @@ scan_screens(Display *dpy)
 	for (i = 0; i < n; i++) {
 
 		char *name = XGetAtomName(dpy, rrm[i].name);
+		Bool must_free_name = True;
 
 		if (name == NULL)
+		{
 			name = "";
+			must_free_name = False;
+		}
 
```

Comment:
Looking at this a bit more, if we can't detect a monitor name coming from the XAtom, then it's an error, and not something we should continue with.  In detecting monitors, we won't ever have stored the empty string to match against, so the rest of the function won't really do a great deal.  I'd rather see:

```
if (name == NULL) {
    fprintf(stderr, "%s: couldn't detect monitor with empty name\n", __func__);
    exit (101);
}
```